extensions/sftp/processors/ListSFTP.cpp

```diff
@@ -186,7 +186,7 @@ ListSFTP::Child::Child(const std::string& parent_path_, std::tuple<std::string /
 }
 
 std::string ListSFTP::Child::getPath() const {
-  return utils::file::FileUtils::concat_path(parent_path, filename, true /*force_posix*/);
+  return utils::StringUtils::join_pack(parent_path, "/", filename);
```

Comment:
This is not used for files, only for URLs, so I wouldn't put it in `FileUtils`.  I can add a `makeUrl()` function in `StringUtils` (or maybe a new `UrlUtils`) which joins its argument with '/'s.  I think that would be less explicit, but it may be more readable.

Reply:
Okay, that makes sense.

Reply:
I added a `StringUtils::joinWithSlashes()` in 4a7c19f078eabede4cfd750fa513b73bc012e654, although TBH I'm not sure if this is better.

Reply:
I'm not a fan of `joinWithSlashes`. I don't think this abstraction helps with readability or performance. `join_pack` with slashes is just fine for URLs IMO, but if you insist on a new abstraction layer, it should probably be a class for URLs with a slash operator, like `std::filesystem::path`.

Reply:
`joinWithSlashes` is almost certainly slower, because `stringstream` is slower than `string::append()`, but I don't think it will be called enough times to make a difference.

Readability-wise, I think both are fine -- we may need a 4th person to decide, as now we have 1 yea, 1 nay, and 1 abstention.

Reply:
My original intention was to extract the previous `concat_path` with `force_posix` option to a separate function for easier access, as it seems to be a common use case. With the last change I don't think `joinWithSlashes` would be a better option, but this functionality as part of a URL utility class is something that sounds better. If that wouldn't be a trivial change then I think that can be delegated to a separate PR.

Reply:
OK, I have reverted 4a7c19f078eabede4cfd750fa513b73bc012e654.

`concat_path(..., force_posix = true)` wasn't used in many places, only in `ListSFTP` (2 times) and `PutSFTP` (3 times).  There may be more opportunities to use a `URL` class, I haven't checked.